File: app/src/main/java/com/githang/gradledoc/common/ListActivity.java
package com.githang.gradledoc.common;

import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;

import com.githang.android.snippet.adapter.BaseListAdapter;
import com.githang.android.snippet.adapter.ItemCreator;
import com.githang.android.snippet.adapter.ItemHolder;
import com.githang.gradledoc.R;

import java.util.List;

/**
 * 列表界面
 *
 * @author 黄浩杭 (huanghaohang@parkingwang.com)
 * @version 2015-12-03
 * @since 2015-12-03
 */
public abstract class ListActivity<T> extends BaseRefreshActivity<List<T>> implements ItemCreator<T, ItemHolder.DefaultHolder>, AdapterView.OnItemClickListener {
    private ListView mListView;
    private BaseListAdapter<T, ItemHolder.DefaultHolder> mAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_list);
        mListView = (ListView) findViewById(android.R.id.list);
        mAdapter = new BaseListAdapter<>(this);
        mListView.setAdapter(mAdapter);
        mListView.setOnItemClickListener(this);
    }

    protected void update(List<T> data) {
        mAdapter.update(data);
    }

    @Override
    public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
    }

    @Override
    public void onHandle(List<T> list) {
        update(list);
    }
}

File: app/src/main/java/com/githang/gradledoc/common/ListActivity.kt
package com.githang.gradledoc.common

import android.os.Bundle
import android.view.View
import android.widget.AdapterView
import android.widget.ListView

import com.githang.android.snippet.adapter.BaseListAdapter
import com.githang.android.snippet.adapter.ItemCreator
import com.githang.android.snippet.adapter.ItemHolder
import com.githang.gradledoc.R

/**
 * 列表界面
 *
 * @author 黄浩杭 (huanghaohang@parkingwang.com)
 * @version 2015-12-03
 * @since 2015-12-03
 */
abstract class ListActivity<T> : BaseRefreshActivity<List<T>>(), ItemCreator<T, ItemHolder.DefaultHolder>, AdapterView.OnItemClickListener {
    private lateinit var listView: ListView
    private lateinit var adapter: BaseListAdapter<T, ItemHolder.DefaultHolder>

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_list)
        listView = findViewById<View>(android.R.id.list) as ListView
        adapter = BaseListAdapter<T, ItemHolder.DefaultHolder>(this)
        listView.adapter = adapter
        listView.onItemClickListener = this
    }

    protected fun update(data: List<T>) {
        adapter.update(data)
    }

    override fun onItemClick(parent: AdapterView<*>, view: View, position: Int, id: Long) {}

    override fun onHandle(data: List<T>) {
        update(data)
    }
}
